test/Identity.FunctionalTests/Extensions/ResponseAssert.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using AngleSharp;
+using AngleSharp.Dom.Html;
+using AngleSharp.Network;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Identity.FunctionalTests
+{
+    public static class ResponseAssert
+    {
+        public static Uri IsRedirect(HttpResponseMessage responseMessage)
+        {
+            Assert.Equal(HttpStatusCode.Redirect, responseMessage.StatusCode);
+            return responseMessage.Headers.Location;
+        }
+
+        public static IHtmlDocument IsHtmlDocument(HttpResponseMessage response)
+        {
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal("text/html", response.Content.Headers.ContentType.MediaType);
+            var content = response.Content.ReadAsStringAsync().Result;
+            var document = BrowsingContext.New()
```

Comment:
Why can't you do `new HtmlParser().Parse` here? The Response headers don't change the result from what I can tell.

Reply:
Because the browsing context sets things like the base Url and things like that. Without it, things like href values will not be correct.